### PR TITLE
[GH-1309] Correct Attachments documentation

### PR DESCRIPTION
**Pull request by Omar8345**
#### Summary

This pull request corrects misinformation in the [attachments documentation](https://developers.mattermost.com/integrate/reference/message-attachments/), in the end of the documentation, it instructs the developer to add an `attachments` key in the `props` field, which won't work, I have edited the documentation and the curl and HTTPS examples to add the `attachments` key in the top-level of the post's JSON.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-developer-documentation/issues/1309

### CODE REVIEW

**Comment by cwarnermm** [comment]
Thanks, @Omar8345, for this developer documentation PR. Your changes will be technically reviewed by Engineering.

**Review by hmhealey — CHANGES_REQUESTED** [review]
Thanks for the PR, @Omar8345! I think there might be some confusion going on here because the syntax in those examples matches how `POST /api/v4/posts` expects the payload to look.

When you tried this, were you perhaps using an incoming webhook? I think the payload sent to a webhook uses the `attachments` field without `props` like in your changes.

If that's the case, I think we could instead add a separate example for posting message attachments using a webhook. If we do that, we should probably also update the existing examples to be posting to `https://{your-mattermost-site}/api/v4/posts` since that's the correct API endpoint for that

**Comment by Omar8345** [comment]
> Thanks for the PR, @Omar8345! I think there might be some confusion going on here because the syntax in those examples matches how `POST /api/v4/posts` expects the payload to look.
> 
> When you tried this, were you perhaps using an incoming webhook? I think the payload sent to a webhook uses the `attachments` field without `props` like in your changes.
> 
> If that's the case, I think we could instead add a separate example for posting message attachments using a webhook. If we do that, we should probably also update the existing examples to be posting to `https://{your-mattermost-site}/api/v4/posts` since that's the correct API endpoint for that

Thanks for your catch, I have made the corresponding changes requested and added a section for how to post a message attachment using a webhook.

File: site/content/integrate/reference/message-attachments/_index.md [hunk 1]
```diff
@@ -139,19 +139,40 @@ And here is how it renders in Mattermost:
...
 Content-Type: application/json
 Content-Length: 192
 
-{"channel_id":"qmd5oqtwoibz8cuzxzg5ekshgr", "message":"Test message #testing", "props":{"attachments": [{"pretext": "This is the attachment pretext.","text": "This is the attachment text."}]}}
+{"channel_id":"qmd5oqtwoibz8cuzxzg5ekshgr", "message":"Test message #testing", "attachments": [{"pretext": "This is the attachment pretext.","text": "This is the attachment text."}]}
+```
+
+### Can I post a message attachment using a webhook?
+
+Yes, you can also use the {{< newtabref href="https://api.mattermost.com/#operation/CreatePost" title="create post RESTful API" >}} to post a message attachment using a webhook.
```
Comment (hmhealey) [inline site/content/integrate/reference/message-attachments/_index.md:161]:
```suggestion
Yes, you can also use an [incoming webhook]({{< ref "/integrate/webhooks/incoming" >}}) to post a message attachment.
```

File: site/content/integrate/reference/message-attachments/_index.md [hunk 2]
```diff
@@ -139,19 +139,40 @@ And here is how it renders in Mattermost:
...
+
+You need to add an `attachments` key to the post's JSON. The value is an array of message attachments you want attached to the post. See below for an example curl command.
+
+`curl -i -X POST -H 'Content-Type: application/json' -d '{"text":"Test message #testing", "attachments": [{"pretext": "This is the attachment pretext.","text": "This is the attachment text."}]}' https://{your-mattermost-site}/api/v4/posts`
```
Comment (hmhealey) [inline site/content/integrate/reference/message-attachments/_index.md:165]:
```suggestion
`curl -i -X POST -H 'Content-Type: application/json' -d '{"text":"Test message #testing", "attachments": [{"pretext": "This is the attachment pretext.","text": "This is the attachment text."}]}' https://{your-mattermost-site}/hooks/{your-webhook-id}`
```

File: site/content/integrate/reference/message-attachments/_index.md [hunk 3]
```diff
@@ -139,19 +139,40 @@ And here is how it renders in Mattermost:
...
+
+Below is an example HTTP request:
+
+```http request
+POST /api/v4/posts HTTP/1.1
```
Comment (hmhealey) [inline site/content/integrate/reference/message-attachments/_index.md:170]:
```suggestion
POST /hooks/{your-webhook-id} HTTP/1.1
```

**Review by hmhealey — APPROVED** [review]
Thanks for adding that! I'm sure I'll find this useful since I've always needed to manually construct the curl command for this when I occasionally need to test something involving message attachments. I've got a couple minor tweaks I want to make, but they're more nitpicky so I can apply them myself

**Review by hmhealey — CHANGES_REQUESTED** [review]
Actually, would you be able to apply those? I guess I don't have the required permission to do that here

**Comment by Omar8345** [comment]
> Actually, would you be able to apply those? I guess I don't have the required permission to do that here

Applied your suggested edits! I was a bit confused 😅

**Comment by cwarnermm** [comment]
@Omar8345 - Can you update your branch with latest master, please? Thanks!

**Comment by Omar8345** [comment]
Done @cwarnermm

**Review by devinbinnie — APPROVED** [review]

**Comment by cwarnermm** [comment]
@hmhealey - This PR is ready for your review.

**Review by hmhealey — APPROVED** [review]
Great catch! I remember mentioning to do that, but I forgot to confirm that it was done in both parts. Thanks for fixing that!

**Pull request merged**